commit a1ae27747f1950e71922563ad3371a5bc74665c7 (refs/changes/67/4167/4)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2018-06-14T10:18:27+01:00 (1 year, 4 months ago)
    
    TRITON-478 agent builds fail as non-root when postinstall.sh writes to /
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/npm/postinstall.sh b/npm/postinstall.sh
index fe6116b..27dd466 100755
--- a/npm/postinstall.sh
+++ b/npm/postinstall.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 if [[ -n "$TRACE" ]]; then
@@ -16,6 +16,11 @@ fi
 set -o errexit
 set -o pipefail
 
+if [[ -z "$npm_config_smfdir" || ! -d "$npm_config_smfdir" ]]; then
+    echo "Skipping config-agent postinstall (assuming build-time install)"
+    exit 0
+fi
+
 DIR=`dirname $0`
 ROOT=$(cd `dirname $0`/../ 2>/dev/null && pwd)
 
